zilliqa/src/consensus.rs

```diff
@@ -802,38 +968,103 @@ impl Consensus {
         Ok(None)
     }
 
-    fn get_next_committee(&mut self) -> Result<Committee> {
-        let mut committee = self.committee()?;
+    fn get_next_committee(&mut self, mut committee: Committee) -> Committee {
+        if committee.is_empty() {
+            panic!("committee is empty, this should never happen");
+        }
+
+        if !self.pending_peers.is_empty() {
+            info!(
+                "*** adding {} pending peers to committee",
+                self.pending_peers.len()
+            );
+        }
+
         committee.add_validators(self.pending_peers.drain(..));
-        Ok(committee)
+
+        if committee.len() <= 1 {
```

Comment:
The check will still print when the length is 1 - this will happen if for example other nodes in the network can never catch up and so it's just the one node producing blocks (I think this is always undesirable outside of tests and bootstrapping the network).

Reply:
Makes sense, right.